util/len-caching-lock/src/rwlock.rs

```diff
@@ -0,0 +1,167 @@
+// Copyright 2015-2018 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+use std::ops::{Deref, DerefMut};
+use std::sync::atomic::AtomicUsize;
+use std::sync::atomic::Ordering;
+
+use parking_lot::{RwLock, RwLockReadGuard, RwLockWriteGuard};
+
+use Len;
+
+/// Can be used in place of a [`RwLock`](../../lock_api/struct.RwLock.html) where 
+/// reading `T`'s `len()` without needing to lock, is advantageous. 
+/// When the WriteGuard is released, `T`'s `len()` will be cached.
+pub struct LenCachingRwLock<T> {
```

Comment:
Same as with `Mutex`, please add `Default` and `Debug`

Reply:
Done